### PR TITLE
Logsession import

### PR DESCRIPTION
**Pull request by CrockAgile**

### CODE REVIEW

File: LogSessionImport/LogSessionImport.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            case LogSessionTransferState::Complete : cout << "Complete" << endl; _application->disconnectPolySync(); break;
+        }
+    }
+
+    int _sessionId;
+
+    LogSessionImport * _importer;
+
+    Application * _application;
+
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:139]: Remove newline

File: LogSessionImport/LogSessionImport.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ *
+ * Log session import example.
+ *
+ * Demonstrates how to use the PolySync log session transfer API to import a previously
+ * exported log session to a new distributed system.
+ *
+ */
+
+#include <iostream>
+#include <PolySyncCore.hpp>
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:37]: Add newline

File: LogSessionImport/LogSessionImport.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#include <PolySyncDataModel.hpp>
+#include <PolySyncApplication.hpp>
+#include <PolySyncLogSessionTransfer.hpp>
+
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:41]: Add newline

File: LogSessionImport/LogSessionImport.cpp [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+using namespace std;
+using namespace polysync;
+using namespace datamodel;
+
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:45]: Add newline

File: LogSessionImport/LogSessionImport.cpp [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+/**
+ * @brief SessionImportExample class
+ */
+class SessionImportExample : public DataSubscriber
+{
+
+public:
+
+    SessionImportExample(
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:54]: Put argument on same line unless extends past 80

File: LogSessionImport/LogSessionImport.cpp [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        connectSubscriberMethod< SessionImportExample >(
+                ApplicationEventMessage::getName(),
+                &SessionImportExample::handleEvent,
+                this );
+
+        _application = Application::getInstance();
+
+        _application->attachSubscriber( this );
+    }
+
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:70]: Add newline ( 2 between access sections )

File: LogSessionImport/LogSessionImport.cpp [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+                // Import can be started with or without a progress callback.
+                // When complete, results will be located at "PSYNC_HOME/rnr_logs/[sessionId]"
+                _importer->start(
+                        this,
+                        &SessionImportExample::handleTransferStatus );
+            }
+        }
+    }
+
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:112]: Two newlines between function definitions

**Comment by shnewto** [comment]
:+1:

File: LogSessionImport/LogSessionImport.cpp [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+            const LogSessionTransferStatus & status )
+    {
+        // Status has other information available as well
+        auto state = status.getState();
+
+        cout << "State: ";
+
+        switch( state )
+        {
+            case LogSessionTransferState::Invalid : cout << "Invalid" << endl; break;
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:123]: We don't have a clear guide for switch statements yet, but the statements should be on new lines.

File: LogSessionImport/LogSessionImport.cpp [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    Application * _application;
+
+};
+
+
+int main( int argc, char ** argv )
+{
+    if( argc != 2 )
+    {
+        cerr << "Usage: " << argv[0] << " [sessionId]" << endl;
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:147]: spaces inside brackets

File: LogSessionImport/LogSessionImport.cpp [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+        return -1;
+    }
+
+    SessionImportExample importExample{ atoi(argv[1]) };
```
Comment (windwardrail) [inline LogSessionImport/LogSessionImport.cpp:151]: spaces inside parens and brackets

**Review by windwardrail — COMMENTED** [review]
Just a bunch of minor C++ style stuff.

**Pull request merged**